api/src/org/apache/cloudstack/api/command/admin/host/ListHostsCmd.java

```diff
@@ -203,9 +201,22 @@ public void execute() {
                 hostResponse.setObjectName("host");
                 hostResponses.add(hostResponse);
             }
-
             response.setResponses(hostResponses, result.second());
         }
+        // Remove sensitive details from host response
+        List<HostResponse> hostResponsesList = response.getResponses();
+        for (HostResponse hostResponse: hostResponsesList) {
+            Map<String, String> hostDetails = hostResponse.getDetails();
+            if (hostDetails == null) continue;
+            if (hostDetails.containsKey("username")) {
+                hostDetails.remove("username");
+            }
+            if (hostDetails.containsKey("password")) {
+                hostDetails.remove("password");
+            }
+            hostResponse.setDetails(hostDetails);
```

Comment:
@bhaisaab, @jburwell, returning password from other apis like reconnectHost etc. isnt required either (api doc response is no different)
I think it should be removed from the lower layer in HostJoinDaoImpl so that no api returns them